### ISSUE TITLE
Add implementation of `annotations` constraint ISL 2.0

### ISSUE THREAD

**Issue by desaikd** [amazon-ion/ion-schema-rust#162]
Add implementation of `annotations` constraint as per ISL 2.0

*Grammar:*
```
<ANNOTATIONS> ::= annotations: <ANNOTATIONS_MODIFIER>... [ <SYMBOL>... ]
                | annotations: <TYPE_REFERENCE>

<ANNOTATIONS_MODIFIER> ::= required::
                         | closed::
```

*Ion Schema Specification:*
https://amazon-ion.github.io/ion-schema/docs/isl-2-0/spec#annotations

**Comment by popematt** [comment]
@desaikd Is this fixed by #175

**Comment by desaikd** [comment]
Yes, this has been fixed with #175.
